fortitude/src/rules/precision/kind_suffixes.rs

```diff
@@ -72,6 +79,46 @@ impl AstRule for NoRealSuffix {
         if !regex_is_match!(r"^(\d*\.\d*|\d*\.*\d*[eE]\d+)$", txt) {
             return None;
         }
+
+        // Ok if being used in a kind statement
+        if let Some(call_expression) = node.ancestors().find(|x| x.kind() == "call_expression") {
```

Comment:
I am slightly worried this is a little too general, as we want exactly the grand-parent (or great-grand-parent for complex literals?) -- but I just searched my whole corpus and the most complicated I saw was stuff like `kind(thing(1))`, so probably this is fine in practice

Reply:
Further down I use `parent()?.parent()?` to do just this, as the first parent is the argument list and the second is the function call. I could use that here too -- I did it this way here in case people used it in cases where the return kind of an expression isn't obvious, like where a C++ programmer might use `decltype`. I suppose in that case, maybe explicit suffixes would be a good idea anyway.

Reply:
I've tightened the search through the ancestors now so the following code will still raise a warning:

```f90
integer, parameter :: sp = kind(1.0 + 1.0)
```